### ISSUE TITLE
Amazon Caching/Wrong Tracking Number Issue

### ISSUE THREAD

**Issue by markspivey** [gbrodman/order-tracking#191]
I've noticed recently that the script seems to be grabbing the wrong tracking numbers (or the wrong tracking link) for Amazon orders. I've had to do some manual reconciliations recently, and if I take the order number and look up the tracking manually, it completely differs from the tracking numbers that are scraped by the script.

I've tried multiple Amazon account logins in my config.yml, and have switched between using Chrome and Firefox, but to no avail.

Is anyone else facing the same issue?

cc: @gbrodman

**Comment by markspivey** [comment]
I don't really understand the behaviour. When I manually copy the Amazon shipping/tracking link from an email, Amazon will redirect me to a random order tracking page that isn't my own.

**Comment by markspivey** [comment]
![image](https://user-images.githubusercontent.com/71787519/116640661-bc715380-a930-11eb-83b3-b47cc2f86a81.png)

If I try and find a tracking number of a personal Amazon account through a business login, it tells me that they can't find the tracking information.

**Comment by gbrodman** [comment]
yep this is a big problem and I'm not sure there's anything we can do about it; it seems like a terrible bug on Amazon's end

**Comment by markspivey** [comment]
> yep this is a big problem and I'm not sure there's anything we can do about it; it seems like a terrible bug on Amazon's end

A massive workaround would be to instead navigate to the order page for the shipped item and click the tracking button from there, which still appears to be working correctly. 

However, I've submitted a shitload of bug reports (and hopefully others have too), so fingers crossed they have this thing fixed quickly.

**Comment by markspivey** [comment]
I believe Amazon have fixed the issue now.

**Comment by markspivey** [comment]
@gbrodman seems like Amazon maybe pushed the change that they initially wanted to implement. It doesn't look like you can use a single Amazon account to grab tracking numbers for any other accounts now.

It looks like we need to sign in with the account that received the ship confirm/tracking details in order to be able to scrape the associated webpage.

**Comment by markspivey** [comment]
@gbrodman seeing as this Amazon change isn't going away, does it make more sense for the email tracking retriever to only scan for emails sent to the active Amazon email address in the config.yml?
